src/OrchardCore.Modules/OrchardCore.Localization/Startup.cs

```diff
@@ -73,4 +76,43 @@ public override void ConfigureServices(IServiceCollection services)
             services.Configure<RequestLocalizationOptions>(options => options.ApplyCurrentCultureToResponseHeaders = true);
         }
     }
+
+    [Feature("OrchardCore.Localization.CulturePicker.Admin")]
+    public class CulturePickerStartup : StartupBase
+    {
+        internal static readonly string AdminSiteCookieName = ".OrchardCore.AdminSiteCulture";
+
+        private static readonly Task<ProviderCultureResult> NullProviderCultureResult = Task.FromResult(default(ProviderCultureResult));
+
+        private readonly AdminOptions _adminOptions;
+
+        public CulturePickerStartup(IOptions<AdminOptions> adminOptions)
+        {
+            _adminOptions = adminOptions.Value;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.Configure<RequestLocalizationOptions>(options => options.AddInitialRequestCultureProvider(new CustomRequestCultureProvider(context =>
+            {
+                if (context.Request.Path.Value[1..].StartsWith(_adminOptions.AdminUrlPrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
Don't we already have a way to check if the current request is for admin? Like an attribute and a static method to check for it?

Reply:
I remembered there's an `AdminAttribute` but I'm not sure about the static method that you are talking about. I will check ...

Reply:
@jtkech I'm not sure sure why `AdminAttribute.IsApplied()` not working as expecting here, any idea?

Reply:
Hmm, I assume your delegate get called by the localization middleware.

If so, this is before the AdminAttribute is applied, which is done in a Resource Filter.

Reply:
> I assume your delegate get called by the localization middleware.

Exactly, which why I stuck here ;) fortunately I suggested [here](https://github.com/OrchardCMS/OrchardCore/pull/12017/files#r922205530) to implement it in a seperate middleware which may resolve the issue then

Reply:
No I don't think so and better to not add an additional middleware.

So here just checking the path is okay for me, hmm but maybe the check is incomplete.

For example if the admin prefix is "blabla", the paths "/blabla", "/balabla/" and "/blabla/subpath" match but "/blablafoo" should not match. So a given path matches if it is equal to "/adminprefix" OR if it starts with "/adminprefix/".

Maybe we are missing the same check when a media file is requested, as I remember we check if the path starts with "/media" where maybe we would need to check if it starts with "/media/".

Reply:
> No I don't think so and better to not add an additional middleware.

But this will fail if someone inject a new `RequestCultureProvider` into `RequestLocalizationOptions`

> Maybe we are missing the same check when a media file is requested, as I remember we check if the path starts with "/media" where maybe we would need to check if it starts with "/media/".

I though the requested media should start with `/media/` not the media management, I will modify the check ..

Reply:
As I expect media management start with `/Admin/media`

Reply:
> As I expect media management start with /Admin/media

I was not talking about this but when serving a media file, e.g. requested by the front-end, where we check for an `AssetsRequestPath` whose default is `DefaultAssetsRequestPath = "/media"`

Reply:
May be I didn't understand your point here, if you mean we need a simialr check, I already updated the PR to check both cases

Reply:
Okay cool then, yes forget the media thing ;) was just a side note.